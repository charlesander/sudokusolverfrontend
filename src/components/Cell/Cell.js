import React, {Component} from 'react';
import PropTypes from 'prop-types'

export default class Cell extends Component {

    static propTypes = {
        cellIndex: PropTypes.number.isRequired,
        cellValue: PropTypes.number.isRequired,
        setCellValue: PropTypes.func.isRequired
    }

    render() {

        return (<div
            className={"cell " + (this.props.cellValue ? "preset" : "settable")}
            key={this.props.cellIndex}
            onClick={this.props.setCellValue.bind(this, this.props.cellIndex)}>{this.props.cellValue}</div>)
    }
}