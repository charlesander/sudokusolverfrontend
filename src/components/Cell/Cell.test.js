import React from "react";
import { create } from "react-test-renderer";
import Cell from "./Cell"

describe("Cell component", () => {
    test("Matches the snapshot", () => {
        let i = 0
        let cellValue = 0
        let setCellValue = () => true
        const cell = create(<Cell
            key={i}
            cellIndex={i}
            cellValue={cellValue}
            setCellValue={setCellValue}/>);
        expect(cell.toJSON()).toMatchSnapshot();
    });
})