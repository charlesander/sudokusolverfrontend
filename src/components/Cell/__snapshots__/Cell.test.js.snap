// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component Matches the snapshot 1`] = `
<div
  className="cell settable"
  onClick={[Function]}
>
  0
</div>
`;
